typing/typecore.ml

```diff
@@ -5377,10 +5394,7 @@ and type_function
         | Some default ->
             assert (is_optional arg_label);
             let ty_default = newvar () in
-            begin
-              try unify env (type_option ty_default) ty_arg_mono
-              with Unify _ -> assert false;
-            end;
+            Result.get_ok (unify env (type_option ty_default) ty_arg_mono);
```

Comment:
Shouldn't this be `assert (Result.is_ok ...)`?

Reply:
As an occasionally `-noassert`-paranoid person, I would not recommend sticking important side-effects inside assertions.

Reply:
My problem `Invalid_argument` raised by `Result.get_ok` will be less help than the original `assert false` in diagnosing any issue with the (undocumented) invariant here